test/unit_tests/braket/default_simulator/openqasm/test_interpreter.py

```diff
@@ -0,0 +1,1473 @@
+        "h",
+        "s",
+        "sdg",
+        "t",
+        "tdg",
+        "sx",
+        "rx",
+        "ry",
+        "rz",
+        "cx",
+        "cy",
+        "cz",
+        "cp",
+        "crx",
+        "cry",
+        "crz",
+        "ch",
+        "swap",
+        "ccx",
+        "cswap",
+        "cu",
+        "CX",
+        "phase",
+        "cphase",
+        "id",
+        "u1",
+        "u2",
+        "u3",
+    }.issubset(context.gate_table.current_scope.keys())
+
+
+def test_adder(adder):
+    circuit = Interpreter().build_circuit("adder.qasm", {"a_in": 1, "b_in": 15}, is_file=True)
+    simulation = StateVectorSimulation(10, 1, 1)
+    simulation.evolve(circuit.instructions)
+    assert simulation.probabilities[0b_0_0001_0000_1] == 1
+
+
+def test_assignment_operators():
+    qasm = """
+    int[16] x;
+    bit[4] xs;
+
+    x = 0;
+    xs = "0000";
+
+    x += 1; // 1
+    x *= 2; // 2
+    x /= 2; // 1
+    x -= 5; // -4
+
+    xs[2:] |= "11";
+    """
+    context = Interpreter().run(qasm)
+    assert context.get_value("x") == IntegerLiteral(-4)
+    assert convert_bool_array_to_string(context.get_value("xs")) == "0011"
+
+
+def test_resolve_result_type():
+    qasm = """
+    float[16] small_pi;
+    float[16] one_point_five;
+    int[8] int_not_bool;
+
+    int[8] one = 1;
+    float[16] half = 0.5;
+    bit t = true;
+
+    small_pi = π + one - 1;
+    one_point_five = one + half;
+    int_not_bool = t + 1;
+    """
+    context = Interpreter().run(qasm)
+    assert context.get_value("small_pi") == FloatLiteral(3.140625)
+    assert context.get_value("one_point_five") == FloatLiteral(1.5)
+    assert context.get_value("int_not_bool") == IntegerLiteral(2)
+
+
+def test_bit_operators():
+    qasm = """
+    bit[4] and;
+    bit[4] or;
+    bit[4] xor;
+    bit[4] lshift;
+    bit[4] rshift;
+    bit[4] flip;
+    bit gt;
+    bit lt;
+    bit ge;
+    bit le;
+    bit eq;
+    bit neq;
+    bit not;
+    bit not_zero;
+
+    bit[4] x = "0101";
+    bit[4] y = "1100";
+
+    and = x & y;
+    or = x | y;
+    xor = x ^ y;
+    lshift = x << 2;
+    rshift = y >> 2;
+    flip = ~x;
+    gt = x > y;
+    lt = x < y;
+    ge = x >= y;
+    le = x <= y;
+    eq = x == y;
+    neq = x != y;
+    not = !x;
+    not_zero = !(x << 4);
+    """
+    context = Interpreter().run(qasm)
+    assert convert_bool_array_to_string(context.get_value("and")) == "0100"
+    assert convert_bool_array_to_string(context.get_value("or")) == "1101"
+    assert convert_bool_array_to_string(context.get_value("xor")) == "1001"
+    assert convert_bool_array_to_string(context.get_value("lshift")) == "0100"
+    assert convert_bool_array_to_string(context.get_value("rshift")) == "0011"
+    assert convert_bool_array_to_string(context.get_value("flip")) == "1010"
+    assert context.get_value("gt") == BooleanLiteral(False)
+    assert context.get_value("lt") == BooleanLiteral(True)
+    assert context.get_value("ge") == BooleanLiteral(False)
```

Comment:
doesn't really test ge is any different from gt and le any different from lt

Reply:
Sure, it doesn't prevent it from being changed in the wrong way, but yes, I'm being too pedantic here.